File: dotNet/Tests/ContextTests.cs
using System.Collections.Generic;
using NUnit.Framework;

namespace Com.Company.Context {
    class ContextTests {
        private IList<ConfigBeanItem> mBeanItemList = new List<ConfigBeanItem> {
            new ConfigBeanItem("id1", typeof(string), "String1"),
            new ConfigBeanItem("id2", typeof(string), "String2"),
            new ConfigBeanItem("id3", typeof(int), 10),
            new ConfigBeanItem("", typeof(double), 0.5),
        };

        private Com.Company.Context.Context mContext;

        [SetUp]
        public void Setup() {
            mContext = new Com.Company.Context.Context(mBeanItemList);
        }

        [Test]
        public void ResolveByType_Test() {
            var instance = mContext.Resolve<double>();
            Assert.AreEqual(instance, 0,5);
        }

        [Test]
        public void ResolveByName_Test() {
            var instance = mContext.Resolve<string>("id1");
            Assert.AreEqual(instance, "String1");
        }

        [Test]
        public void ResolveAmbiguous_Test() {
            try {
                mContext.Resolve<string>();
                Assert.Fail("Ambiguous are not allowed");
            } catch (ContextException) {
            }
        }

        [Test]
        public void AbsentBean_Test() {
            try {
                mContext.Resolve<object>();
                Assert.Fail("Ambiguous are not allowed");
            } catch (ContextException) {
            }
        }
    }
}

File: java/tests/com/company/context/ContextTests.java
package com.company.context;

import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

class ContextTests {
    private ConfigBeanItem[] mBeanItemArrays = new ConfigBeanItem[] {
            new ConfigBeanItem("id1", String.class, "String1"),
            new ConfigBeanItem("id2", String.class, "String2"),
            new ConfigBeanItem("id3", int.class, 10),
            new ConfigBeanItem("", double.class, 0.5),
    };

    private Context mContext = new Context(Arrays.asList(mBeanItemArrays));

    @Test
    void resolveByType_Test() {
        try {
            double instance = mContext.resolve(double.class);
            assertEquals(instance, 0,5);
        } catch (Exception ex) {
            fail("Exception occurs: " + ex.getMessage());
        }
    }

    @Test
    void resolveByName_Test() {
        try {
            String instance = mContext.resolve(String.class, "id1");
            assertEquals(instance, "String1");
        } catch (Exception ex) {
            fail("Exception occurs: " + ex.getMessage());
        }
    }

    @Test
    void resolveAmbiguous_Test() {
        try {
            mContext.resolve(String.class);
            fail("Ambiguous are not allowed");
        } catch (Exception ex) {
            assertEquals(ex.getClass(), ContextException.class);
        }
    }

    @Test
    void absentBean_Test() {
        try {
            mContext.resolve(Object.class);
            fail("Ambiguous are not allowed");
        } catch (Exception ex) {
            assertEquals(ex.getClass(), ContextException.class);
        }
    }
}
